app/controllers/admin/timelogs_controller.rb

```diff
@@ -0,0 +1,77 @@
+module Admin
+  require 'trello' 
+  class Admin::TimelogsController < ApplicationController
+    before_action :authenticate_user!
+    before_action :load_timelog, only: [:update, :destroy]
+    before_action :load_trello_service, only: [:index, :new]
+
+    def index
+      if params[:filter]
+        @timelogs = User.find(params[:user_id]).timelogs.paginate(page: params[:page], per_page: 5).send params[:filter] 
```

Comment:
what we can do here:
1. `User.find(params[:user_id])`. This code we can find in all actions. So it is better to move it to a separate method and have an instance variable.
2. We can have a method called `filtered_timelogs` which is going to be responsible only for filtering.
```
def filtered_timelogs
  @user.timelogs.send params[:filter]
end
```
P.S. using send is not the best, but let it be for now :)
3. We can paginate based on format. If it is pdf, we don't need to paginate, just use `filtered_timelogs`. If it is `html` we'll paginate: `filtered_timelogs.paginate...`